Bottom: 676a245422a93f850b73eb5f53bedcbf869be122
Top:    7143593195ffe7e2f53621bfff39bc2b40b3376f
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-07 13:54:07 +0800

autofs-5.0.7 - fix potential null dereference in lookup_mount()

Updating a negative cache entry should always find an entry but the entry
lookup return isn't checked and probably should be.


---

diff --git a/modules/lookup_file.c b/modules/lookup_file.c
index 2836996..58663c6 100644
--- a/modules/lookup_file.c
+++ b/modules/lookup_file.c
@@ -1131,7 +1131,7 @@ do_cache_lookup:
 				       mapent, ctxt->parse->context);
 	if (ret) {
 		time_t now = time(NULL);
-		int rv = CHE_OK;
+		int rv = CHE_FAIL;
 
 		cache_writelock(mc);
 		me = cache_lookup_distinct(mc, key);
diff --git a/modules/lookup_ldap.c b/modules/lookup_ldap.c
index a59de92..d89c295 100644
--- a/modules/lookup_ldap.c
+++ b/modules/lookup_ldap.c
@@ -3012,7 +3012,7 @@ int lookup_mount(struct autofs_point *ap, const char *name, int name_len, void *
 				       mapent, ctxt->parse->context);
 	if (ret) {
 		time_t now = time(NULL);
-		int rv = CHE_OK;
+		int rv = CHE_FAIL;
 
 		/* Record the the mount fail in the cache */
 		cache_writelock(mc);
diff --git a/modules/lookup_sss.c b/modules/lookup_sss.c
index 5c2ed0a..013d3b4 100644
--- a/modules/lookup_sss.c
+++ b/modules/lookup_sss.c
@@ -673,7 +673,7 @@ int lookup_mount(struct autofs_point *ap, const char *name, int name_len, void *
 				       mapent, ctxt->parse->context);
 	if (ret) {
 		time_t now = time(NULL);
-		int rv = CHE_OK;
+		int rv = CHE_FAIL;
 
 		/* Record the the mount fail in the cache */
 		cache_writelock(mc);
diff --git a/modules/lookup_yp.c b/modules/lookup_yp.c
index a716e1f..72a8baa 100644
--- a/modules/lookup_yp.c
+++ b/modules/lookup_yp.c
@@ -699,7 +699,7 @@ int lookup_mount(struct autofs_point *ap, const char *name, int name_len, void *
 					       mapent, ctxt->parse->context);
 		if (ret) {
 			time_t now = time(NULL);
-			int rv = CHE_OK;
+			int rv = CHE_FAIL;
 
 			cache_writelock(mc);
 			me = cache_lookup_distinct(mc, key);
